src/query/storage/converter.go

```diff
@@ -127,34 +198,60 @@ func PromTimeSeriesToSeriesAttributes(series prompb.TimeSeries) (ts.SeriesAttrib
 		promMetricType = ts.PromMetricTypeStateSet
 
 	default:
-		return ts.SeriesAttributes{}, fmt.Errorf("invalid Prometheus metric type %v", series.Type)
+		return ts.SeriesAttributes{}, fmt.Errorf("invalid metric type for Open Metrics: %s", series.Type)
+	}
+
+	m3MetricType, err := convertM3Type(series.M3Type)
+	if err != nil {
+		return ts.SeriesAttributes{}, err
 	}
 
+	return ts.SeriesAttributes{
+		Source:            ts.SourceTypeOpenMetrics,
+		PromType:          promMetricType,
+		M3Type:            m3MetricType,
+		HandleValueResets: handleValueResets,
+	}, nil
+}
+
+func seriesAttributesForGraphiteSource(series prompb.TimeSeries) (ts.SeriesAttributes, error) {
+	m3MetricType, err := convertM3Type(series.M3Type)
+	if err != nil {
+		return ts.SeriesAttributes{}, err
+	}
+
+	var promMetricType ts.PromMetricType
 	switch series.M3Type {
 	case prompb.M3Type_M3_COUNTER:
-		m3MetricType = ts.M3MetricTypeCounter
-		if promMetricType == ts.PromMetricTypeUnknown && series.Source == prompb.Source_GRAPHITE {
-			promMetricType = ts.PromMetricTypeCounter
-		}
+		promMetricType = ts.PromMetricTypeCounter
 	case prompb.M3Type_M3_GAUGE:
-		m3MetricType = ts.M3MetricTypeGauge
-		if promMetricType == ts.PromMetricTypeUnknown && series.Source == prompb.Source_GRAPHITE {
-			promMetricType = ts.PromMetricTypeGauge
-		}
-	case prompb.M3Type_M3_TIMER:
-		m3MetricType = ts.M3MetricTypeTimer
-	default:
-		return ts.SeriesAttributes{}, fmt.Errorf("invalid M3 metric type %v", series.M3Type)
+		promMetricType = ts.PromMetricTypeGauge
```

Comment:
no default case anymore?

Reply:
This is covered by the call to `convertM3Type` above.
The switch here only affects `promMetricType`.